header-only/examples/cython/_demo.pyx

```diff
@@ -0,0 +1,24 @@
+from demo_impl cimport ArrayBuffers, create_demo_array as create_demo_array_impl
+from cython.operator cimport dereference as deref
+
+# Import both types and functions
+cimport numpy as np
+import numpy as np
+import awkward as ak
+
+
+def create_demo_array():
+    array_buffers = create_demo_array_impl()
+
+    buffers = {}
+    # Convert from raw buffers to NumPy arrays
+    for it in array_buffers.buffers:
+        # Pull out name, buffer, and lookup length of buffer
+        name = it.first
+        buffer = it.second
+        nbytes = array_buffers.buffer_nbytes[name]
+
+        # Store buffer in Python dict
+        buffers[name.decode('UTF-8')] = np.asarray(<np.uint8_t[:nbytes]>buffer)
```

Comment:
@agoose77 - does it transfer ownership of the C++ `ArrayBuffers` buffers?

Reply:
Oooh :facepalm:!! Not sure how _that_ got lost between writing the `pybind11` example and the `cython` one. I love the review process :sweat_smile:

Reply:
or `PyArray_ENABLEFLAGS(buffers[...], np.NPY_OWNDATA)`?

Reply:
> or `PyArray_ENABLEFLAGS(buffers[...], np.NPY_OWNDATA)`?

So, I'm not an expert here, but what I've read so far implies that NumPy would use its own allocator to free the memory if one uses `NPY_OWNDATA`. I *think* they map onto `malloc` and `free`, but I think being explicit in using the cython view that definitely calls `free` might be safer?

Reply:
> > or `PyArray_ENABLEFLAGS(buffers[...], np.NPY_OWNDATA)`?
> 
> So, I'm not an expert here, but what I've read so far implies that NumPy would use its own allocator to free the memory if one uses `NPY_OWNDATA`. I _think_ they map onto `malloc` and `free`, but my gut is that being explicit in using the cython view that definitely calls `free` might be safer?

Looking into the cython code: it looks like it also uses `malloc` and `free`. Perhaps, allocating the `ArrayBuffers` buffers should be also done with `malloc`. `new` allocated memory is on a `Free Store` and `malloc` one is on the `Heap` - technically it could be the same.

Reply:
Yes, that's a good point. I was willing to sweep _that_ one under the rug, but we should be proper here!